pkg/virt-launcher/virtwrap/api/converter.go

```diff
@@ -1794,6 +1767,34 @@ func indexNetworksByName(networks []v1.Network) (map[string]*v1.Network, error)
 	return netsByName, nil
 }
 
+func createSRIOVHostDevice(hostPCIAddress string, guestPCIAddress string, bootOrder *uint) (HostDevice, error) {
+	var hostDev HostDevice
+
+	hostAddr, err := decoratePciAddressField(hostPCIAddress)
+	if err != nil {
+		return hostDev, err
+	}
+	hostDev = HostDevice{
+		Source:  HostDeviceSource{Address: hostAddr},
+		Type:    "pci",
+		Managed: "no",
+	}
+
+	if guestPCIAddress != "" {
+		addr, err := decoratePciAddressField(guestPCIAddress)
+		if err != nil {
+			return hostDev, err
```

Comment:
Why do you return the `hostDev1? I would return `nil, err`.

Reply:
`HostDevice` is not a pointer here. You suggest to return a pointer instead of the structure?

Reply:
yes

Reply:
Done